### PR TITLE
Finish tex{Sub}Image*D related spec.

### PR DESCRIPTION
**Pull request by zhenyao**
Please take a look.

### CODE REVIEW

**Comment by zhenyao** [comment]
@jdashg Please review. Thanks.

File: specs/latest/2.0/index.html [hunk 1]
```diff
@@ -1307,34 +1307,80 @@ <h2 class="no-toc">Table of contents</h2>
...
       </dt>
       <dd>
-          Allocates and initializes the specified mipmap level of a three-dimensional or
-          two-dimensional array texture. <br><br>
-
-          If <code>pixels</code> is null, a buffer of sufficient size initialized to 0 is
-          passed. <br><br>
+        <p>Only differences from <a href="https://www.khronos.org/registry/webgl/specs/latest/1.0/#TEXIMAGE2D">texImage2D</a> in WebGL 1 are described here. </p>
+        <p>Sized internal formats are supported in WebGL 2 and <em>internalformat</em> is no longer required to be the same as <em>format</em>. Instead, the combination of <em>internalformat</em>, <em>format</em>, and <em>type</em> must be listed in Tabel 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a></span>.</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Typo: Tabel -> Tabel

**Comment by zhenyao** [comment]
Fixed typo.

File: specs/latest/2.0/index.html [hunk 2]
```diff
@@ -1307,34 +1307,80 @@ <h2 class="no-toc">Table of contents</h2>
...
-          inherited from the WebGL 1.0 API.</div>
+      <dt class="idl-code">void texImage2D(GLenum target, GLint level, GLenum internalformat, GLenum format, GLenum type, TexImageSource? source) /* May throw DOMException */
+        <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_full_spec_3.0.3.pdf#nameddest=section-3.8.3">OpenGL ES 3.0.3 &sect;3.8.4</a>, <a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a>)</span>
+      </dt>
+      <dd>
+        <p>Only differences from <a href="https://www.khronos.org/registry/webgl/specs/latest/1.0/#TEXIMAGE2D_HTML">texImage2D</a> in WebGL 1 are described here. </p>
+        <p>Sized internal formats are supported in WebGL 2 and <em>internalformat</em> is no longer required to be the same as <em>format</em>. Instead, the combination of <em>internalformat</em>, <em>format</em>, and <em>type</em> must be listed in Tabel 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a></span>.</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Typo: Tabel -> Table

File: specs/latest/2.0/index.html [hunk 3]
```diff
@@ -1307,34 +1307,80 @@ <h2 class="no-toc">Table of contents</h2>
...
+      </dd>
 
-          If an attempt is made to call this function with no WebGLTexture bound (see above), an
-          <code>INVALID_OPERATION</code> error is generated. <br><br>
+      <dt class="idl-code"><a name="TEXSUBIMAGE2D">void texSubImage2D</a>(GLenum target, GLint level, GLint xoffset, GLint yoffset, GLsizei width, GLsizei height, GLenum format, GLenum type, ArrayBufferView? pixels)
+        <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_full_spec_3.0.3.pdf#nameddest=section-3.8.5">OpenGL ES 3.0.3 &sect;3.8.5</a>, <a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexSubImage2D.xhtml">man page</a>)</span>
+      </dt>
+      <dd>
+        <p>Only differences from <a href="https://www.khronos.org/registry/webgl/specs/latest/1.0/#TEXSUBIMAGE2D">texSubImage2D</a> in WebGL 1 are described here. </p>
+        <p>The combination of <em>format</em>, <em>type</em>, and WebGLTexture's internal format must be listed in Tabel 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a></span>.</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Tabel

File: specs/latest/2.0/index.html [hunk 4]
```diff
@@ -1307,34 +1307,80 @@ <h2 class="no-toc">Table of contents</h2>
...
+        <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_full_spec_3.0.3.pdf#nameddest=section-3.8.3">OpenGL ES 3.0.3 &sect;3.8.4</a>, <a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage3D.xhtml">man page</a>)</span>
+      </dt>
+      <dd>
+        <p>Allocates and initializes the specified mipmap level of a three-dimensional or two-dimensional array texture. </p>
+        <p>If <code>pixels</code> is null, a buffer of sufficient size initialized to 0 is passed. </p>
+        <p>If <code>pixels</code> is non-null and its size is less than what is required by the specified <em>width</em>, <em>height</em>, <em>depth</em>, <em>format</em>, and <em>type</em>, generates an <code>INVALID_VALUE</code> error.</p>
+        <p>The combination of <em>internalformat</em>, <em>format</em>, and <em>type</em> must be listed in Tabel 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage3D.xhtml">man page</a></span>.</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Tabel

File: specs/latest/2.0/index.html [hunk 5]
```diff
@@ -1344,21 +1390,13 @@ <h2 class="no-toc">Table of contents</h2>
...
-
-        <div class="note editor">Needs update to determine restrictions for <em>type</em></div>
-
-        If <code>pixels</code> is null then an </code>INVALID_VALUE</code> error is generated. <br><br>
-
-        See <a href="../1.0/#PIXEL_STORAGE_PARAMETERS">Pixel Storage Parameters</a> for WebGL-specific
-        pixel storage parameters that affect the behavior of this function.
+        <p>Update a rectangular subregion of the currently bound WebGLTexture. </p>
+        <p>If an attempt is made to call this function with no WebGLTexture bound (see above), generates an <code>INVALID_OPERATION</code> error. </p>
+        <p>The combination of <em>format</em>, <em>type</em>, and WebGLTexture's internal format must be listed in Tabel 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage3D.xhtml">man page</a></span>.</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Tabel

**Comment by kenrussell** [comment]
LGTM overall. Please update the last modified date of the spec at the top of the file.

@jdashg : do you have any comment on this? Would like to merge it now.

**Comment by zhenyao** [comment]
Date updated.

**Comment by kenrussell** [comment]
Very nice. LGTM

**Comment by zhenyao** [comment]
@jdashg  ping

**Comment by kenrussell** [comment]
Merging now.

**Comment by zhenyao** [comment]
Rebased

**Comment by kenrussell** [comment]
Merging now.

**Pull request merged**